rust/lance/src/dataset/sql.rs

```diff
@@ -63,18 +63,20 @@ impl SqlQueryBuilder {
     }
 
     pub async fn build(self) -> lance_core::Result<SqlQuery> {
-        let ctx = SessionContext::new();
+        let ctx = get_session_context(&LanceExecutionOptions::default());
         let row_id = self.with_row_id;
         let row_addr = self.with_row_addr;
         ctx.register_table(
-            self.table_name,
+            self.table_name.clone(),
             Arc::new(LanceTableProvider::new(
                 self.dataset.clone(),
                 row_id,
                 row_addr,
             )),
         )?;
         let df = ctx.sql(&self.sql).await?;
+
+        ctx.deregister_table(self.table_name)?;
```

Comment:
The session context got from ```get_session_context``` is cloned from a static session context, and they share  the registered tables. So we need to deregister it after query.

There is still a chance we might ran into 'register table' conflicts. Maybe we should use a new session context each time?